### PR TITLE
Maintenance week updates

### PR DESCRIPTION
**Pull request by ehanson8**
### Purpose and background context
Updates app according to our [maintenance week documentation](https://mitlibraries.atlassian.net/wiki/spaces/IN/pages/3445948428/DataEng+Application+Maintenance).

This the first app we've updated to 3.12 and there were no significant issues.

### How can a reviewer manually see the effects of these changes?
Run `make test` and `make lint` to confirm they still pass

### Includes new or updated dependencies?
YES

### Changes expectations for external applications?
YES

### Developer
- [x] All new ENV is documented in README
- [x] All new ENV has been added to staging and production environments
- [x] All related Jira tickets are linked in commit message(s)
- [x] Stakeholder approval has been confirmed (or is not needed)

### Code Reviewer(s)
- [ ] The commit message is clear and follows our guidelines (not just this PR message)
- [ ] There are appropriate tests covering any new functionality
- [ ] The provided documentation is sufficient for understanding any new functionality introduced
- [ ] Any manual tests have been performed and verified
- [ ] New dependencies are appropriate or there were no changes

### CODE REVIEW

**Review by ghukill — APPROVED** [review]
Looks good to me!  After some local `pipenv` wrangling (discussed on slack), installation and tests and linting ran great.

**Review by jonavellecuerdo — APPROVED** [review]
Changes were straightforward and look good to me! ✨ Just had one non-blocking question.

File: .python-version [hunk 1]
```diff
@@ -1 +1 @@
-3.11.4
\ No newline at end of file
+3.12.2
```
[the remark this answers is not in the document]
Comment (ehanson8) [reply .python-version:1]: I've just gone with what minor version is installed when running `pyenv install 3.12`, in this case it was `3.12.2`. @ghukill may have a more deliberate way of doing this

[the remark this answers is not in the document]
Comment (ghukill) [reply .python-version:1]: I'm ashamed to say I don't.

I would probably check [python.org downloads page](https://www.python.org/downloads/) to get a sense of what the newest patch version is (defining "patch" as the third digit in `major.minor.patch` [semantic versioning](https://semver.org/) schema).  Then, like you say @ehanson8, see what kind of options come back from `pyenv` which will likely be doing the installation.  Then weigh the two.

Another data point is the `Dockerfile` image version.  I see this uses `public.ecr.aws/lambda/python:3.12`, but again we don't know precisely (without some digging) to know which pathc version this is.  

All that said, where possible, I think it's good to strive for the most recent patch version within the target major.minor version.  But, weigh this against the reality our Dockerfiles, `pyenv`, and `asdf`, among others may thwart those best attempts.

You know... leaving all that thinking out loud text above for context, but in reality, we might not want to specify the patch version, unless we know something specific about why it's needed.  That way it's major.minor (e.g. `3.12`) and all the tooling will get the most up-to-date patch version.  I think this more accurately reflects that it's just a bit messy at that level, depending on the tooling and context.

[the remark this answers is not in the document]
Comment (ehanson8) [reply .python-version:1]: Ugh, right, it's the patch version not minor version. Can we get away with just `3.12` in `.python-version`? I've just always included that without questioning it but you bring up a good point

[the remark this answers is not in the document]
Comment (ghukill) [reply .python-version:1]: I'd reckon we could just go with major + minor (e.g. `3.12`) and chalk up the patch version to whatever we get, we get.  

If _anything_, we'd probably want to aim for the patch version that the Dockerfile builds.  But, don't think it's worth the effort, given it's a different OS / architecture, so at that level, possible error parity still isn't gauranteed.  

Yep, I think major + minor is sufficient.

[the remark this answers is not in the document]
Comment (ehanson8) [reply .python-version:1]: OK, I'll do `3.12`, remove and rebuild the virtual env to confirm nothing breaks, and then merge

[the remark this answers is not in the document]
Comment (ehanson8) [reply .python-version]: Nothing broke! Merging

**Pull request merged**